docs/_pages/typecheck.md

```diff
@@ -188,6 +188,40 @@ local strings: Array<string> = {"Hello", "world!"}
 local numbers: Array<number> = {1, 2, 3, 4, 5, 6}
 ```
 
+## Generic functions
+
+As well as generic type aliases like `Array<T>`, Luau supports generic functions. These are functions that, as well as their regular data parameters, take type parameters. For example, a function which reverses an array is:
+```
+function reverse(a)
+  local result = {}
+  for i = #a, 1, -1 do
+    table.insert(result, a[i])
+  end
+  return result
+end
+```
+The type of this function is that it can reverse an array, and return an array of the same type. Luau can infer this type, but if you want to be explicit, you can declare the type parameter `T`, for example:
+```
+function reverse<T>(a: Array<T>): Array<T>
+  local result: Array<T> = {}
```

Comment:
We have builtin syntax for array types.  We should probably use that syntax here so that people aren't misled into believing that they should define an `Array<T>` alias on their own.
```suggestion
function reverse<T>(a: {T}): {T}
  local result: {T} = {}
```

Reply:
I'm just cut-and-pasting from the section above.